src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx

```diff
@@ -1,126 +1,76 @@
+  orgId: string;
+  tagKey: string;
+};
+
+type Props = {
+  group: Group;
+  location: {
+    query: object;
   };
-
-  componentDidMount() {
-    this.fetchData();
-  }
-
-  componentDidUpdate(prevProps) {
-    const queryHasChanged = !isEqual(prevProps.location.query, this.props.location.query);
-    if (queryHasChanged || prevProps.params.tagKey !== this.props.params.tagKey) {
-      this.fetchData();
-    }
-  }
-
-  fetchData = async () => {
-    const {
-      params,
-      location: {query},
-    } = this.props;
-    this.setState({
-      loading: true,
-      error: false,
-    });
-
-    const promises = [
-      this.props.api.requestPromise(`/issues/${params.groupId}/tags/${params.tagKey}/`, {
-        query,
-      }),
-      this.props.api.requestPromise(
-        `/issues/${params.groupId}/tags/${params.tagKey}/values/`,
-        {
-          query,
-          includeAllArgs: true,
-        }
-      ),
+} & RouteComponentProps<RouteParams, {}>;
+
+type State = {
+  tagKey: Tag;
+  tagValueList: TagValue[];
+  tagValueListPageLinks: string;
+};
+
+class GroupTagValues extends AsyncComponent<
+  Props & AsyncComponent['props'],
+  State & AsyncComponent['state']
+> {
+  getEndpoints(): [string, string][] {
+    const {groupId, tagKey} = this.props.params;
+    return [
+      ['tagKey', `/issues/${groupId}/tags/${tagKey}/`],
+      ['tagValueList', `/issues/${groupId}/tags/${tagKey}/values/`],
     ];
+  }
 
-    try {
-      const [tagKey, tagValueResponse] = await Promise.all(promises);
-      const [tagValueList, , jqXHR] = tagValueResponse;
-
-      this.setState({
-        tagKey,
-        tagValueList,
-        loading: false,
-        pageLinks: jqXHR.getResponseHeader('Link'),
-      });
-    } catch (rejections) {
-      // eslint-disable-next-line no-console
-      console.error(rejections);
-      this.setState({
-        error: true,
-        loading: false,
-      });
-    }
-  };
-
-  getUserDisplayName(item) {
+  getUserDisplayName(item): string {
     return item.email || item.username || item.identifier || item.ipAddress || item.value;
   }
 
-  render() {
-    if (this.state.loading) {
-      return <LoadingIndicator />;
-    } else if (this.state.error) {
-      return <LoadingError onRetry={this.fetchData} />;
-    }
-
+  renderBody() {
     const {
       group,
       params: {orgId},
     } = this.props;
-    const tagKey = this.state.tagKey;
-
-    const sortedTagValueList = sortBy(
-      this.state.tagValueList,
+    const {tagKey, tagValueList, tagValueListPageLinks} = this.state;
+    const sortedTagValueList: TagValue[] = sortBy(
+      tagValueList,
       property('count')
     ).reverse();
 
     const issuesPath = `/organizations/${orgId}/issues/`;
 
-    const children = sortedTagValueList.map((tagValue, tagValueIdx) => {
-      const pct = percent(tagValue.count, tagKey.totalValues).toFixed(2);
+    const children = sortedTagValueList.map((tagValue: TagValue, tagValueIdx: number) => {
```

Comment:
I believe that `sortedTagValueList` should already come with the types `tagValue: TagValue, tagValueIdx: number`  and it's not necessary to define them once again

Reply:
fixed!